src/components/AuthModal/AuthModal-container.tsx

```diff
@@ -38,6 +41,19 @@ const AuthModal = ({
     setOpen(prevState => prevState && !auth.isAuth);
   }, [auth.isAuth]);
 
+  const node = useRef<HTMLDivElement>(null);
+
+  const handleMenuState = () => {
+    setMenuVisibility(!menuVisibility)
+  }
```

Comment:
jeżeli bazujesz na poprzednim stanie to warto w setState użyć funkcji `setState(prevState => {})` dzięki temu masz gwarancje poprzedniego stanu